src/server/shell/decoration/basic_manager.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright © 2019 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 or 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored By: William Wold <william.wold@canonical.com>
+ */
+
+#ifndef MIR_SHELL_DECORATION_BASIC_MANAGER_H_
+#define MIR_SHELL_DECORATION_BASIC_MANAGER_H_
+
+#include "manager.h"
+#include <functional>
+#include <unordered_map>
+#include <mutex>
+
+namespace mir
+{
+class Executor;
+namespace graphics
+{
+class GraphicBufferAllocator;
+}
+namespace shell
+{
+class Shell;
+namespace decoration
+{
+class Decoration;
+/// Draws Mir's default window decorations and provides basic move, resize, close, etc functionality
```

Comment:
This doesn't (at least not yet) do any of the above. The intention is for it to do so in future? Or will this do the "management" and another class do the drawing, gesture support, etc.?